www/js/con-author.js

```diff
@@ -148,7 +190,55 @@ define([
 		 */
 		handleVariableName: function(name){
 			console.log("**************** in handle Variable Name ", name);
-
+			/* check if node with name already exists and
+			 if name is parsed as valid variable
+			 */
+			var nameID = this._model.authored.getNodeIDByName(name);
+			// If nameID is falsy give "null"; if it doesn't match, give "false"
+			console.log("nameID, currentID",nameID,this.currentID,equation.isVariable(name));
+			this.applyDirectives(this.authorPM.process(nameID?!(nameID==this.currentID):null,'variableName',name, equation.isVariable(name)));
+			
+			//var logObj = {};
+			if(!this._model.authored.getNodeIDByName(name) && equation.isVariable(name)){
+				// check all nodes in the model for equations containing name of this node
+				// replace name of this node in equation with its ID
+				this._model.active.setName(this.currentID, name);
+				this.updateNodes();
+				//not required - because updateNodes() will add connections automatically
+				//this.setConnections(this._model.active.getInputs(this.currentID), this.currentID);
+				// update equation labels is not required for quantity variable name handling , verify once
+				//this.updateEquationLabels();
+				
+				logObj = {
+					error: false
+				};
+				
+			} else {
+				//logging the error case
+				logObj = {
+					error: true
+				};
+				if(this._model.authored.getNodeIDByName(name)){
+					lang.mixin(logObj, {
+						message: "duplication"
+					});
+				} else if(equation.isVariable(name)){
+					lang.mixin(logObj, {
+						message: "parse"
+					});
+				}
+			}
+			this.enableDisableSetStudentNode();
+			
+			
+			logObj = lang.mixin({
+				type: "solution-enter",
+				nodeID: this.currentID,
+				propoerty: "name",
+				node: name,
+				value: name
+			}, logObj);
+			//this.logging.log('solution-step', logObj);
```

Comment:
we need to either find a common place where we can call this logging object or do something else. Otherwise we will keep adding it to every handler and then that would be hard to maintain in the future due to redundancy.